pydoctor/epydoc/markup/epytext.py

```diff
@@ -1352,69 +1352,101 @@ def to_stan(self, docstring_linker: DocstringLinker) -> Tag:
+            assert len(children)==1
+            # The contents is encapsulated inside a section node. 
+            # Reparent the contents of the second level to the root level. 
+            self._document.children.extend(set_nodes_parent(children[0].children, self._document))
+        
+        return self._document
+    
+    def _to_node(self, tree: Union[Element, str]) -> Iterable[nodes.Node]:
+        
         if isinstance(tree, str):
-            return tree
-
-        if tree.tag == 'section':
-            seclevel += 1
+            yield set_node_attributes(nodes.Text(tree), document=self._document)
+            return
 
-        # Process the variables first.
-        variables = [self._to_stan(c, linker, seclevel) for c in tree.children]
+        # Process the children first.
+        variables: List[nodes.Node] = []
+        for c in tree.children:
+            variables.extend(self._to_node(c))
 
         # Perform the approriate action for the DOM tree type.
         if tree.tag == 'para':
-            if tree.attribs.get('inline'):
-                return variables
-            else:
-                return tags.p(*variables)
+            # we yield a paragraph node even if tree.attribs.get('inline') is True because
+            # the choice to render the <p> tags is handled in _PydoctorHTMLTranslator.should_be_compact_paragraph(), not here anymore
+            yield set_node_attributes(nodes.paragraph('', ''), document=self._document, children=variables)
         elif tree.tag == 'code':
-            return tags.code(*variables)
+            yield set_node_attributes(nodes.literal('', ''), document=self._document, children=variables)
         elif tree.tag == 'uri':
-            return tags.a(variables[0], href=variables[1], target='_top')
+            _label, _target = variables[0], variables[1]
+            yield set_node_attributes(nodes.reference(
+                    '', internal=False, refuri=_target), document=self._document, children=_label.children)
+        
         elif tree.tag == 'link':
-            label, target = variables
-            lineno = int(cast(Element, tree.children[1]).attribs['lineno'])
-            return linker.link_xref(target, label, lineno)
+            _label, _target = variables
+            assert isinstance(_target, nodes.Text)
+            assert isinstance(_label, nodes.inline)
+
+            args = {}
+            if _target.astext() != _label.astext():
+                args['refuri']=_target.astext()
+            
+            # Figure the line number to warn on precise lines. 
+            # This is needed only for links currently.
+            lineno: Optional[int] = None
+            elem = tree.children[1]
+            if isinstance(elem, Element):
+                lineno = int(elem.attribs['lineno'])
+
+            yield set_node_attributes(nodes.title_reference(
+                   '', '', **args), document=self._document, lineno=lineno, children=_label.children)
```

Comment:
Unlike the old code, we no longer call `link_xref()` here. This means that linking is now done when converting the docutils document to Stan. Is that the best time to resolve crosslinks or would it be better to do it here and change the interface of `DocstringLinker` to use docutils types instead?

Reply:
One advantage of resolving early would be that we could skip HTML generation entirely if we want to implement a "lint mode" that checks docstrings without producing output.

Reply:
>  Is that the best time to resolve crosslinks or would it be better to do it here and change the interface of DocstringLinker to use docutils types instead?

This is probably something that could be left for a next PR. Also it has been working this way forever in the the `restructuredtext.py` module. So this change of behaviour makes us more consistent. 

If the goal is only to provide a "lint mode", then we could probably do it with a little extra code, that do not required us to touch this "sensible" part of the `epytex.py` module. But instead process the `ParsedDocstring` of all documentables in the systems, and report errors appropriately.

Reply:
Indeed the `restructuredtext` code doesn't resolve links, so maybe doing it very early isn't a good idea. It would be good to resolve links before generating HTML though. I'm fine with that being done in a separate PR.

Reply:
The `link_xref()` call is done in our `HTMLTranslator` when converting docutils documents to stan. 

I suppose the right thing to do would be to use a custom Transform and apply it to all docstring documents. But this means that instead of having a `ParsedDocstring.to_node() -> nodes.document` method, we should provide a `ParsedDocstring.document: nodes.document` attribute such that we can transform it in place. I guess it could work with the `to_node()` method  too(since it essentially returns the self._document attribute, bu that would be a bit confusing for future maintenance.). Also the thing to consider is that, currently not all `ParsedDocstring` subclasses implements `to_node()`. Namely the `AnnotationDocstring` and the `ParsedPlaintextDocstring`. `AnnotationDocstring` will be removed in #402, and we discussed `ParsedPlaintextDocstring` already.

Reply:
So maybe we can create a separate ticket to take care of that ?

Reply:
Done #421